--- tmp/dolfin/dolfin/cmake/modules/FindParMETIS.cmake.orig	2013-08-14 00:22:41.567100613 -0700
+++ tmp/dolfin/dolfin/cmake/modules/FindParMETIS.cmake	2013-08-14 00:26:55.283857802 -0700
@@ -41,13 +41,21 @@
     DOC "Directory where the ParMETIS header files are located"
   )
 
+  find_path(METIS_INCLUDE_DIRS metis.h
+    HINTS ${METIS_DIR}/include ${PARMETIS_DIR}/include 
+          $ENV{METIS_DIR}/include $ENV{PARMETIS_DIR}/include
+          ${PETSC_DIR}/include
+    DOC "Directory where the METIS header files are located"
+  )
   find_library(PARMETIS_LIBRARY parmetis
     HINTS ${PARMETIS_DIR}/lib $ENV{PARMETIS_DIR}/lib ${PETSC_DIR}/lib
     DOC "Directory where the ParMETIS library is located"
   )
 
   find_library(METIS_LIBRARY metis
-    HINTS ${PARMETIS_DIR}/lib $ENV{PARMETIS_DIR}/lib ${PETSC_DIR}/lib
+    HINTS ${METIS_DIR}/lib ${PARMETIS_DIR}/lib
+          $ENV{METIS_DIR}/lib $ENV{PARMETIS_DIR}/lib 
+          ${PETSC_DIR}/lib
     DOC "Directory where the METIS library is located"
   )
 
@@ -64,7 +72,7 @@
   elseif (PARMETIS_INCLUDE_DIRS AND PARMETIS_LIBRARY)
 
   # Set flags for building test program
-  set(CMAKE_REQUIRED_INCLUDES  ${PARMETIS_INCLUDE_DIRS} ${MPI_CXX_INCLUDE_PATH})
+  set(CMAKE_REQUIRED_INCLUDES  ${PARMETIS_INCLUDE_DIRS} ${METIS_INCLUDE_DIRS} ${MPI_CXX_INCLUDE_PATH})
   set(CMAKE_REQUIRED_LIBRARIES ${PARMETIS_LIBRARIES}    ${MPI_CXX_LIBRARIES})
   set(CMAKE_REQUIRED_FLAGS     ${CMAKE_REQUIRED_FLAGS}  ${MPI_CXX_COMPILE_FLAGS})
 
